// viser-react/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Area", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Area;
  }
});
Object.defineProperty(exports, "Axis", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Axis;
  }
});
Object.defineProperty(exports, "Bar", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Bar;
  }
});
Object.defineProperty(exports, "Box", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Box;
  }
});
Object.defineProperty(exports, "Brush", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Brush;
  }
});
Object.defineProperty(exports, "Candle", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Candle;
  }
});
Object.defineProperty(exports, "Contour", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Contour;
  }
});
Object.defineProperty(exports, "Coord", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Coord;
  }
});
Object.defineProperty(exports, "DashLine", {
  enumerable: true,
  get: function get() {
    return _SubComponent.DashLine;
  }
});
Object.defineProperty(exports, "DodgeBar", {
  enumerable: true,
  get: function get() {
    return _SubComponent.DodgeBar;
  }
});
Object.defineProperty(exports, "DodgeInterval", {
  enumerable: true,
  get: function get() {
    return _SubComponent.DodgeInterval;
  }
});
Object.defineProperty(exports, "Edge", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Edge;
  }
});
Object.defineProperty(exports, "Funnel", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Funnel;
  }
});
Object.defineProperty(exports, "Guide", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Guide;
  }
});
Object.defineProperty(exports, "Heatmap", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Heatmap;
  }
});
Object.defineProperty(exports, "Interval", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Interval;
  }
});
Object.defineProperty(exports, "JitterPoint", {
  enumerable: true,
  get: function get() {
    return _SubComponent.JitterPoint;
  }
});
Object.defineProperty(exports, "Legend", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Legend;
  }
});
Object.defineProperty(exports, "Line", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Line;
  }
});
Object.defineProperty(exports, "Path", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Path;
  }
});
Object.defineProperty(exports, "Pie", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Pie;
  }
});
Object.defineProperty(exports, "Point", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Point;
  }
});
Object.defineProperty(exports, "Polygon", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Polygon;
  }
});
Object.defineProperty(exports, "Pyramid", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Pyramid;
  }
});
Object.defineProperty(exports, "Sankey", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Sankey;
  }
});
Object.defineProperty(exports, "Schema", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Schema;
  }
});
Object.defineProperty(exports, "Sector", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Sector;
  }
});
Object.defineProperty(exports, "Series", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Series;
  }
});
Object.defineProperty(exports, "SmoothArea", {
  enumerable: true,
  get: function get() {
    return _SubComponent.SmoothArea;
  }
});
Object.defineProperty(exports, "SmoothLine", {
  enumerable: true,
  get: function get() {
    return _SubComponent.SmoothLine;
  }
});
Object.defineProperty(exports, "StackArea", {
  enumerable: true,
  get: function get() {
    return _SubComponent.StackArea;
  }
});
Object.defineProperty(exports, "StackBar", {
  enumerable: true,
  get: function get() {
    return _SubComponent.StackBar;
  }
});
Object.defineProperty(exports, "StackInterval", {
  enumerable: true,
  get: function get() {
    return _SubComponent.StackInterval;
  }
});
Object.defineProperty(exports, "StackLine", {
  enumerable: true,
  get: function get() {
    return _SubComponent.StackLine;
  }
});
Object.defineProperty(exports, "Tooltip", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Tooltip;
  }
});
Object.defineProperty(exports, "Venn", {
  enumerable: true,
  get: function get() {
    return _SubComponent.Venn;
  }
});
Object.defineProperty(exports, "Slider", {
  enumerable: true,
  get: function get() {
    return _SubPlugin.Slider;
  }
});
Object.defineProperty(exports, "Chart", {
  enumerable: true,
  get: function get() {
    return _Chart.default;
  }
});
Object.defineProperty(exports, "View", {
  enumerable: true,
  get: function get() {
    return _View.default;
  }
});
Object.defineProperty(exports, "FacetView", {
  enumerable: true,
  get: function get() {
    return _FacetView.default;
  }
});
Object.defineProperty(exports, "Facet", {
  enumerable: true,
  get: function get() {
    return _Facet.default;
  }
});
Object.defineProperty(exports, "LiteChart", {
  enumerable: true,
  get: function get() {
    return _LiteChart.default;
  }
});
Object.defineProperty(exports, "Plugin", {
  enumerable: true,
  get: function get() {
    return _plugin.default;
  }
});
exports.Global = exports.registerShape = exports.registerAnimation = void 0;

var viser = _interopRequireWildcard(require("viser"));

var _SubComponent = require("./components/SubComponent");

var _SubPlugin = require("./plugins/SubPlugin");

var _Chart = _interopRequireDefault(require("./components/Chart"));

var _View = _interopRequireDefault(require("./components/View"));

var _FacetView = _interopRequireDefault(require("./components/FacetView"));

var _Facet = _interopRequireDefault(require("./components/Facet"));

var _LiteChart = _interopRequireDefault(require("./components/LiteChart"));

var _plugin = _interopRequireDefault(require("./plugins/plugin"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var registerAnimation = viser.registerAnimation;
exports.registerAnimation = registerAnimation;
var registerShape = viser.registerShape;
exports.registerShape = registerShape;
var Global = viser.Global;
exports.Global = Global;
